test/simple_malloc.cc

```
@@ -177,99 +177,96 @@ int main(int argc, char *argv[]) {
+      ->default_val(0)
+      ->check(CLI::NonNegativeNumber);
 
 #ifdef USE_DD_PROFILING
-    bool start_profiling = false;
-    app.add_flag("--profile", start_profiling, "Enable profiling")
-        ->default_val(false);
+  bool start_profiling = false;
+  app.add_flag("--profile", start_profiling, "Enable profiling")
+      ->default_val(false);
 #endif
 
-    CLI11_PARSE(app, argc, argv);
+  CLI11_PARSE(app, argc, argv);
 
 #ifdef USE_DD_PROFILING
-    if (start_profiling && ddprof_start_profiling() != 0) {
-      fprintf(stderr, "Failed to start profiling\n");
-      return 1;
-    }
+  if (start_profiling && ddprof_start_profiling() != 0) {
+    fprintf(stderr, "Failed to start profiling\n");
+    return 1;
+  }
 #endif
-    if (exec_args.empty()) {
-      print_header();
-    }
+  if (exec_args.empty()) {
+    print_header();
+  }
 
-    for (unsigned int i = 1; i < nb_forks; ++i) {
-      if (fork()) {
-        break;
-      }
+  for (unsigned int i = 1; i < nb_forks; ++i) {
+    if (fork()) {
+      break;
     }
+  }
 
-    if (!exec_args.empty()) {
-      std::vector<char *> new_args;
-      for (auto &a : exec_args) {
-        new_args.push_back(a.data());
-      }
-      execvp(new_args[0], new_args.data());
-      perror("Exec failed: ");
-      return 1;
+  if (!exec_args.empty()) {
+    std::vector<char *> new_args;
+    for (auto &a : exec_args) {
+      new_args.push_back(a.data());
     }
+    execvp(new_args[0], new_args.data());
+    perror("Exec failed: ");
+    return 1;
+  }
 
-    std::vector<std::thread> threads;
-    std::vector<Stats> stats{nb_threads};
-    for (unsigned int i = 1; i < nb_threads; ++i) {
-      threads.emplace_back(wrapper, std::cref(opts), std::ref(stats[i]));
-    }
-    wrapper(opts, stats[0]);
-    for (auto &t : threads) {
-      t.join();
-    }
-    auto pid = getpid();
-    for (auto &stat : stats) {
-      print_stats(pid, stat);
-    }
-  } catch (...) { return 1; }
```

@nsavoire I did not see the point of the try / catch (I was actually confused as to why this was failing), are you OK removing this ?

---

I think it's a good practice to catch all exceptions in main, here what was missing was printing errors I guess.
What was the error ?

---

As discussed, leaving exception propagate, makes the crash more explicit.